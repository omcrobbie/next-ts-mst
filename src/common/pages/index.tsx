import * as React from 'react';
import { inject, observer } from 'mobx-react';

interface PropTypes {
  lastUpdate: string;
  stop: Function;
  start: Function;
}
@observer
export class Index extends React.Component<PropTypes> {
  componentDidMount() {
    this.props.start();
  }
  render() {
    const { lastUpdate, stop } = this.props;
    return (
        <div>
          {lastUpdate.toString()}
          <p>Hello Next.js</p>
          <button onClick={() => stop()}>Stop timer</button>
        </div>
    )

  }

}
  
  export default inject(({store}) => ({
    lastUpdate: store.lastUpdate,
    stop: store.stop,
    start: store.start
  }))(Index);