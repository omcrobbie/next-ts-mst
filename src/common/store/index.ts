import { applySnapshot } from "mobx-state-tree";
import { Store, IStore } from './models';
import { createContext } from 'react';

let store: IStore;

const initStore = (isServer, snapshot = null) => {
  if (isServer) {
    store = Store.create({ lastUpdate: Date.now() });
  }
  if (store as any === null) {
    store = Store.create({ lastUpdate: Date.now() });
  }
  if (snapshot) {
    applySnapshot(store, snapshot);
  }
  return store;
};

export const getStore = () => {
  return createContext(store);
}
export default initStore;
