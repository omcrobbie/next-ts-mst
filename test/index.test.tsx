import { shallow, render } from 'enzyme';
import { Index } from '../src/common/pages';

describe('Index tests', () => {
    it('should render', () => {
        const props = {
            start: () => null,
            stop: () => null,
            lastUpdate: new Date('2/5/2019').toString()
        }
        const wrapped = render(<Index {...props} />);
        expect(wrapped).toMatchSnapshot();
    })
})