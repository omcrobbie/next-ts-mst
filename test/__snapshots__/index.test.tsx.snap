// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index tests should render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "Tue Feb 05 2019 00:00:00 GMT-0500 (EST)",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Hello Next.js",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Stop timer",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Hello Next.js",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Stop timer",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "data": "Tue Feb 05 2019 00:00:00 GMT-0500 (EST)",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Stop timer",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Hello Next.js",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "data": "Tue Feb 05 2019 00:00:00 GMT-0500 (EST)",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
